src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,260 @@
+                  "Climb Mech Ligament",
+                  1,
+                  Radians.of(climberSimNormal.getAngleRads()).in(Degrees)));
+      climbMechTargetLigament =
+          climbMechRoot.append(
+              new MechanismLigament2d(
+                  "Climb Mech Target Ligament",
+                  0.8,
+                  climberTargetAngle.in(Degrees),
+                  10,
+                  new Color8Bit(255, 255, 0)));
+      SmartDashboard.putData("Climb Mech", climbMech);
+    }
+  }
+
+  public Command zero() {
+    return runOnce(
+        () -> {
+          climberMotorFront.setPosition(
+              Degrees.of((climberFrontEncoder.get() + 90) * ClimberConstants.KGearRatio)
+                  .in(Rotations));
+          climberMotorBack.setPosition(
+              Degrees.of((climberBackEncoder.get() + 90) * ClimberConstants.KGearRatio)
+                  .in(Rotations));
+        });
+  }
+
+  public Command runRaw(Voltage voltage) {
+    return runOnce(
+        () -> {
+          climberMotorFront.setControl(new VoltageOut(voltage));
+          climberMotorBack.setControl(new VoltageOut(voltage.times(-1)));
+        });
+  }
+
+  private Command runClimber(Angle position, int slot) {
+    return runOnce(
+        () -> {
+          climberTargetAngle = position;
+          climberMotorFront.setControl(
+              new PositionVoltage(position.times(ClimberConstants.KGearRatio)).withSlot(slot));
+          climberMotorBack.setControl(
+              new PositionVoltage(position.times(ClimberConstants.KGearRatio)).withSlot(slot));
+          SmartDashboard.putNumber("Climber Position Setpoint", position.in(Degrees));
+        });
+  }
+
+  private BooleanSupplier isClimberAtPosition(Angle position) {
+    return () ->
+        position.isNear(
+            climberMotorFront.getPosition().getValue().div(ClimberConstants.KGearRatio),
+            ClimberConstants.kClimberSensorError.div(ClimberConstants.KGearRatio));
+  }
+
+  public Command climb() {
+    return Commands.sequence(
+        runClimber(ClimberConstants.kClimberAtCageSetpoint, 0),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberAtCageSetpoint)),
+        runClimber(ClimberConstants.kClimberExtendedSetpoint, 1));
+  }
+
+  public Command retract() {
+    return runClimber(ClimberConstants.kClimberRetractedSetpoint, 0);
+  }
+
+  private SingleJointedArmSim getSimulator() {
+    if (isClimbing) {
+      return climberSimLifting;
+    } else {
+      return climberSimNormal;
+    }
+  }
+
+  private void toggleClimbingSim() {
+    if (isClimbing) {
+      climberSimNormal.setState(
+          climberSimLifting.getAngleRads(), climberSimLifting.getVelocityRadPerSec());
+      isClimbing = false;
+    } else {
+      climberSimLifting.setState(
+          climberSimNormal.getAngleRads(), climberSimNormal.getVelocityRadPerSec());
+      isClimbing = true;
+    }
+    SmartDashboard.putBoolean("Climbing", isClimbing);
+  }
```

Comment:
Why is this a simulation only thing? don't we need to know when the state changes to change the PID on the real robot?
Seems like this should just be a toggleClimb() function that does something different if robot is real or is sim.